src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -8,17 +8,19 @@
 using Microsoft.AspNetCore.Components.Web.HtmlRendering;
 using Microsoft.AspNetCore.Html;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 using static Microsoft.AspNetCore.Internal.LinkerFlags;
 
 namespace Microsoft.AspNetCore.Components.Endpoints;
 
 internal partial class EndpointHtmlRenderer
 {
     private static readonly object ComponentSequenceKey = new object();
+    private bool stopAddingTasks;
```

Comment:
This needs to be prefixed with `_`. I'm surprised the build hasn't failed on this.
```suggestion
    private bool _stopAddingTasks;
```